SpatialGDK/Source/SpatialGDK/Private/SpatialView/ViewCoordinator.cpp

```diff
@@ -42,9 +58,66 @@ void ViewCoordinator::SendRemoveComponent(Worker_EntityId EntityId, Worker_Compo
 	View.SendRemoveComponent(EntityId, ComponentId);
 }
 
-const TArray<CreateEntityResponse>& ViewCoordinator::GetCreateEntityResponses() const
+Worker_RequestId ViewCoordinator::SendReserveEntityIdsRequest(uint32 NumberOfEntityIds, TOptional<uint32> TimeoutMillis)
+{
+	View.SendReserveEntityIdsRequest({NextRequestId, NumberOfEntityIds, TimeoutMillis});
+	return NextRequestId++;
+}
+
+Worker_RequestId ViewCoordinator::SendCreateEntityRequest(TArray<ComponentData> EntityComponents,
+	TOptional<Worker_EntityId> EntityId, TOptional<uint32> TimeoutMillis)
+{
+	View.SendCreateEntityRequest({NextRequestId, MoveTemp(EntityComponents), EntityId, TimeoutMillis});
+	return NextRequestId++;
+}
+
+Worker_RequestId ViewCoordinator::SendDeleteEntityRequest(Worker_EntityId EntityId, TOptional<uint32> TimeoutMillis)
+{
+	View.SendDeleteEntityRequest({NextRequestId, EntityId, TimeoutMillis});
+	return NextRequestId++;
+}
+
+Worker_RequestId ViewCoordinator::SendEntityQueryRequest(EntityQuery Query, TOptional<uint32> TimeoutMillis)
+{
+	View.SendEntityQueryRequest({NextRequestId, MoveTemp(Query), TimeoutMillis});
+	return NextRequestId++;
+}
+
+Worker_RequestId ViewCoordinator::SendEntityCommandRequest(Worker_EntityId EntityId, CommandRequest Request,
+	TOptional<uint32> TimeoutMillis)
+{
+	View.SendEntityCommandRequest({EntityId, NextRequestId, MoveTemp(Request), TimeoutMillis});
+	return NextRequestId++;
+}
+
+void ViewCoordinator::SendEntityCommandResponse(Worker_RequestId RequestId, CommandResponse Response)
+{
+	View.SendEntityCommandResponse({RequestId, MoveTemp(Response)});
+}
+
+void ViewCoordinator::SendEntityCommandFailure(Worker_RequestId RequestId, FString Message)
+{
+	View.SendEntityCommandFailure({RequestId, MoveTemp(Message)});
```

Comment:
Is MoveTemp here this simply an efficiency thing?

Reply:
Yes. Pass the message by value then move the result. It's the standard thing to do when you need to own a copy and you don't want to write multiple overloads.